### PR TITLE
Master pos diff on bank payment methods jcb

### PR DESCRIPTION
**Pull request by caburj**
Forward port of https://github.com/odoo/odoo/pull/77351

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/point_of_sale/models/pos_session.py [hunk 1]
```diff
@@ -1327,14 +1404,26 @@ def show_journal_items(self):
...
 
+    def _get_other_related_moves(self):
+        # TODO This is not an ideal way to get the diff account.move's for
+        # the session. It would be better if there is a relation field where
+        # these moves are saved.
+
+        # Unfortunately, the 'ref' of account.move is not indexed, so
+        # we are querying over the account.move.line because its 'ref' is indexed.
+        # And yes, we are only concern for split bank payment methods.
+        diff_lines_ref = map(lambda pm: self._get_diff_account_move_ref(pm), self.payment_method_ids.filtered(lambda pm: pm.type == 'bank' and pm.split_transactions))
```
Comment (pimodoo) [inline addons/point_of_sale/models/pos_session.py:1415]: what you wanted to do is:

[self._get_diff_account_move_ref(pm) for pm in self.payment_method_ids if pm.type == 'bank' and pm.split_transactions]

It'll automatically return a list, so you won't have to call *diff_lines_ref next line

Probably we should put this change in 15.0 too, but first merge this master branch

Comment (caburj) [reply addons/point_of_sale/models/pos_session.py:1415]: how about the use of '{}' as default params?

**Comment by pimodoo** [comment]
@caburj for the default value {} sorry it is my fault, I made you change, but it seems our style doesn't let it pass: https://stackoverflow.com/questions/26320899/why-is-the-empty-dictionary-a-dangerous-default-value-in-python

**Comment by pimodoo** [comment]
@robodoo r+ rebase-ff

**Pull request closed without merging**